experimental/filesystem-backends.mdx

```diff
@@ -0,0 +1,332 @@
+---
+title: Filesystem Backends
+description: Details on the experimental filesystem backends for Flipt.
+---
+
+<Note>This feature was added in Flipt `v1.23.0`.</Note>
+
+The following backend types are designed to support declarative management of feature flag state.
+In particular, they're designed to support GitOps practices with minimal external dependencies.
+
+This feature can be enabled via Flipts configuration parameters when running the `flipt` server binary.
+As always, these parameters can be expressed via environment variables or Flipt's yaml configuration file:
+
+<Tabs>
+  <Tab title="Environment Variable">
+
+    ```bash
+    FLIPT_EXPERIMENTAL_FILESYSTEM_STORAGE_ENABLED=true
+    ```
+
+  </Tab>
+  <Tab title="Configuration Yaml">
+
+    ```yaml
+    experimental:
+      filesystem_storage:
+        enabled: true
+    ```
+
+  </Tab>
+</Tabs>
+
+## Backend Types
+
+There are currently two backend types we classify as being filesystem backends, `local` and `git`.
+The `local` backend has been primarily developed to support a local development experience.
+Whereas, the `git` backend is intended for production and serves flag state directly from a configurable repository and Git reference (e.g. branch).
+
+### `local` type
+
+The purpose of this backend type is to support serving Flipt flag state directly from your local filesystem.
+You can simply specify a relative or absolute directory in order to start a local Flipt instance and serve flag state.
+This is particularly useful for local development and validation of flag state changes.
+
+Flipt will periodically rebuild its state from the local disk.
```

Comment:
Maybe we can mention what this is by default. I do not believe this is configurable currently, and maybe I should make it as such?

Reply:
Yeah it would be good to make that configurable., and good shout I will mention the default somewhere here.